cmd/fpga_admissionwebhook/fpga_admissionwebhook.go

```diff
@@ -190,11 +190,11 @@ func serve(w http.ResponseWriter, r *http.Request, admit admitFunc) {
 
 	resp, err := json.Marshal(response)
 	if err != nil {
-		fmt.Println("ERROR:", err)
+		klog.Warningf("%+v", err)
```

Comment:
We use `%+v` to print wrapped errors (the ones augmented with a stack trace). The rule is that if an error is returned by "our" function then it's always wrapped. So, we see immediately where exactly the error was triggered.

The errors returned by external functions like `json.Marshal()` are not wrapped naturally. It's enough call `klog.Warning(err)` here.

Reply:
Also if a response cannot be marshaled then it must be an error in our code, not input data. Probably this should be `klog.Error(err)`?

Reply:
OK. moved to `Error(err)`.

Reply:
good comments @rojkov. Perhaps we should make a note about these in `DEVEL.md`?

Reply:
Yes, should be in `DEVEL.md`. It's just that I haven't spent enough time to explain the same thing  over and over again :)